zio-kafka-test/src/test/scala/zio/kafka/consumer/internal/RunloopSpec.scala

```diff
@@ -52,6 +54,38 @@ object RunloopSpec extends ZIOSpecDefault {
           )
         }
       },
+      test(
+        "runloop does not starts a new stream for partition which being revoked right after assignment within the same RebalanceEvent"
+      ) {
+        withRunloop { (mockConsumer, partitionsHub, runloop) =>
+          mockConsumer.schedulePollTask { () =>
+            mockConsumer.updateEndOffsets(Map(tp10 -> Long.box(0L), tp12 -> Long.box(0L)).asJava)
+            mockConsumer.rebalance(Seq(tp10, tp11).asJava)
+            mockConsumer.rebalance(Seq(tp10).asJava)
+          }
+          mockConsumer.schedulePollTask { () =>
+            mockConsumer.rebalance(Seq(tp10, tp12).asJava)
+            mockConsumer.addRecord(makeConsumerRecord(tp10, key123))
+            mockConsumer.addRecord(makeConsumerRecord(tp12, key123))
+          }
```

Comment:
Why do we need the third rebalance?
Would this work?
```suggestion
          mockConsumer.schedulePollTask { () =>
            mockConsumer.updateEndOffsets(Map(tp10 -> Long.box(0L), tp11 -> Long.box(0L)).asJava)
            mockConsumer.rebalance(Seq(tp10, tp11).asJava)
            mockConsumer.rebalance(Seq(tp10).asJava)
            mockConsumer.addRecord(makeConsumerRecord(tp10, key123))
          }
```

Reply:
This will work in case of verification based on `Diagnostics`, as you mentioned in https://github.com/zio/zio-kafka/pull/1294#discussion_r1718438415, 'cause we will have Rebalance event, which states that only one partition assigned (and the revoked one is not in the assigned set), but with the current state of the test, we cannot guarantee that stream for partition `tp11` is not started without having "something that happened after" (start of partition `tp12`, 'cause otherwise we don't know if `tp11` is not gonna be started at all, or just not started yet, and adding an awaiting timeout here seems like a bad tone to me).

Reply:
Right I see your point, but adding mock behavior for a second poll is not going to help, the stream for tp11 could still be starting when the second poll is finished. Anyways, the point is moot because we have a good alternative.